src/plugins/intel_gpu/src/plugin/ops/pad.cpp

```diff
@@ -18,17 +18,47 @@ static void CreatePadOp(Program& p, const std::shared_ptr<ngraph::op::v1::Pad>&
     auto inputs = p.GetInputInfo(op);
     std::string layerName = layer_type_name_ID(op);
 
+    std::vector<cldnn::input_info> parameter_inputs = {inputs[0]};
+    int32_t non_constant_input_mask = 0;
+
+    auto pads_begin_constant = std::dynamic_pointer_cast<ngraph::op::v0::Constant>(op->input_value(1).get_node_shared_ptr());
+    std::vector<int64_t> pads_begin = std::vector<int64_t>{};
+    if (pads_begin_constant) {
+        pads_begin = pads_begin_constant->cast_vector<int64_t>();
+    } else {
+        parameter_inputs.push_back(inputs[1]);
+        non_constant_input_mask |= cldnn::border::PAD_NON_CONST_INPUT::BEGIN;
+    }
+
+    auto pads_end_constant = std::dynamic_pointer_cast<ngraph::op::v0::Constant>(op->input_value(2).get_node_shared_ptr());
+    std::vector<int64_t> pads_end = std::vector<int64_t>{};
+    if (pads_end_constant) {
+        pads_end = pads_end_constant->cast_vector<int64_t>();
+    } else {
+        parameter_inputs.push_back(inputs[2]);
+        non_constant_input_mask |= cldnn::border::PAD_NON_CONST_INPUT::END;
+    }
+
     float pad_value = 0.f;
-    if (op->get_input_size() == 4) {
+    bool is_value_const = false;
+    if (op->get_pad_mode() == ov::op::PadMode::CONSTANT && op->get_input_size() == 4) {
         auto const_node = std::dynamic_pointer_cast<ngraph::op::v0::Constant>(op->get_input_node_shared_ptr(3));
-        OPENVINO_ASSERT(const_node, "Unsupported const node type in ", op->get_friendly_name(), " (", op->get_type_name(), ")");
-        ov::op::util::get_single_value(const_node, pad_value);
+        if (const_node) {
+            ov::op::util::get_single_value(const_node, pad_value);
+            is_value_const = true;
+        }
+    }
+
+    if (!is_value_const) {
+        parameter_inputs.push_back(inputs[3]);
+        non_constant_input_mask |= cldnn::border::PAD_NON_CONST_INPUT::VALUE;
     }
 
     auto tilePrim = cldnn::border(layerName,
-                                  inputs[0],
-                                  op->get_pads_begin(),
-                                  op->get_pads_end(),
+                                  parameter_inputs,
```

Comment:
"parameter_inputs" are not proper naming I think. Because "ov::Parameter" node is the name of "network input layer" of the ov graph. From the point of the general cldnn node, there is no concept about "parameter". I think just "inputs" is okay for here.

Reply:
Changed to "non_constant_inputs" to be different from origin "inputs"